### PR TITLE
Countdown rework

### PR DESCRIPTION
**Pull request by Purplo-cf**
Top-level changes for countdown rework. Unity elements that are no longer necessary have been removed

### CODE REVIEW

File: Assets/Script/Gameplay/HUD/CountdownDisplay.cs [hunk 1]
```diff
@@ -14,48 +14,27 @@ public class CountdownDisplay : MonoBehaviour
...
+        private Image _progressBar;
 
         [Space]
         [SerializeField]
         private CanvasGroup _canvasGroup;
 
         private Coroutine _currentCoroutine;
 
-        private int _measuresLeft;
+        private bool _displayState;
```
Comment (RileyTheFox) [inline Assets/Script/Gameplay/HUD/CountdownDisplay.cs:25]: I think this boolean could use a slightly more clear name personally. I had to look at where the value is set to understand what it was. Something like `_displayActive` would be better I think

File: Assets/Script/Gameplay/HUD/CountdownDisplay.cs [hunk 2]
```diff
@@ -69,11 +48,13 @@ public void ForceReset()
 
         private void ToggleDisplay(bool newState)
```
Comment (RileyTheFox) [inline Assets/Script/Gameplay/HUD/CountdownDisplay.cs:49]: If going through with the state variable rename, this should also be `bool active` or something.

**Review by RileyTheFox — COMMENTED** [review]
Besides the small comments, looks good. I think for some extra customisation we could have a setting to change the display into a seconds countdown instead of measures? I know some people prefer a countdown in seconds, and Clone Hero has this option too, so it would be nice to have here.

Display setting should default to measures and have the option to change it to seconds.

**Comment by RileyTheFox** [comment]
@EliteAsian123 @kaduwaengertner what do you think regarding the suggestion above?

**Comment by EliteAsian123** [comment]
I'm always for customization 👍

**Comment by kaduwaengertner** [comment]
I like the idea

**Comment by Purplo-cf** [comment]
The new setting and the requested fixes are in. This will need the latest `countdown-rework-core` changes to run

**Comment by EliteAsian123** [comment]
Ah shoot wait, I'm dumb. I made some changes to the countdown prefab on `engine-refactor` because I didn't see that this PR changes those. 

Essentially I just made sure to change all `Transform`s to `RectTransform`s on the prefab and removed the scaling on the transforms as that's bad practice. I'll revert the changes on the main branch to prevent potentially broken prefabs from merging.

**Comment by Purplo-cf** [comment]
No worries! I did a rebase this morning to double check that everything still looked okay. Okay had to make a few small adjustments

**Review by RileyTheFox — CHANGES_REQUESTED** [review]
Some issues I've noticed:

Something has corrupted and the lyric bar is just a white box now. I got a warning in Unity when checking out the branch "Broken text PPtr in file". It was referencing fileID `6635242913724598267` which is found as the m_Father in some places in CountdownDisplay.prefab

image

Also in practice mode, if you select a section that occurs after a countdown earlier in the song, it will show the last part of the countdown. However this is a general engine issue with practice mode that has only come up now (queued updates being generated from time 0 up to the practice section time) so I will handle this at a later point.

**Review by RileyTheFox — APPROVED** [review]

**Pull request merged**